dask/diagnostics/progress.py

```diff
@@ -93,6 +103,8 @@ def _finish(self, dsk, state, errored):
         sys.stdout.write('\n')
         sys.stdout.flush()
 
+        self.last_duration = elapsed
+
```

Comment:
If `elapsed < self._minimum` then this won't get set. Perhaps Move it above the first if statement.